### PR TITLE
Allow Normal distribution with zero std deviation

### PR DESCRIPTION
**Pull request by simonbyrne**
Part of #283. I have hit this issue a few times: it can be very handy, especially for simulation.

See also https://github.com/JuliaStats/StatsFuns.jl/pull/62, which tweaks the underlying functions to be more useful.

### CODE REVIEW

**Comment by matbesancon** [comment]
Should we add a test for the behavior of `Normal` with `std = 0.0`?

**Comment by simonbyrne** [comment]
Yes, but probably need the StatsFuns change merged first.

**Comment by matbesancon** [comment]
Also, the documentation should mention the behavior when continuous distributions are degenerate (mass point)

**Comment by mschauer** [comment]
#62 is merged in the meantime

File: src/univariate/continuous/normal.jl [hunk 1]
```diff
@@ -27,7 +27,10 @@ struct Normal{T<:Real} <: ContinuousUnivariateDistribution
     μ::T
     σ::T
 
-    Normal{T}(μ, σ) where {T} = (@check_args(Normal, σ > zero(σ)); new{T}(μ, σ))
+    function Normal{T}(μ, σ) where {T}
```
Comment (matbesancon) [inline src/univariate/continuous/normal.jl]: Mention the behavior in the documentation of `Normal`

Comment (matbesancon) [reply src/univariate/continuous/normal.jl]: also, some tests for the std=0 case should be added

**Comment by matbesancon** [comment]
@mschauer #62 is an issue closed in 2013, you meant something else?

File: src/univariate/continuous/normal.jl (hunk 1 above)
Comment (simonbyrne) [reply src/univariate/continuous/normal.jl]: Fixed!

**Comment by simonbyrne** [comment]
Any final comments before I merge?

File: REQUIRE [hunk 2]
```diff
@@ -1,6 +1,6 @@
 julia 0.7
 PDMats 0.9.0
-StatsFuns 0.3.1
+StatsFuns 0.7.1
```
Comment (matbesancon) [inline REQUIRE]: why require up to patch number? Why not 0.7?

Comment (simonbyrne) [reply REQUIRE]: 0.7.1 was the release with the necessary changes (in hindsight it should have been a minor release, I apologise).

Comment (matbesancon) [reply REQUIRE]: ok, we can let everything so for this release, in a next one it would be good to remove patch from the other packages requirements

**Pull request merged**